### PR TITLE
FileTarget: Fix archiving for ArchiveFileName without a pattern.

### PR DESCRIPTION
**Pull request by bhaeussermann**
Fixes issue #1108

### CODE REVIEW

File: src/NLog/Targets/FileTarget.cs [hunk 1]
```diff
@@ -524,6 +538,23 @@ protected internal string NewLineChars
             }
         }
 
+        private void RefreshFileArchive()
+        {
+            var dummyEvent = new LogEventInfo();
```
Comment (304NotModified) [inline src/NLog/Targets/FileTarget.cs]: We have `LogEventInfo.CreateNullEvent()` for this :)

File: src/NLog/Targets/FileTarget.cs [hunk 2]
```diff
@@ -1493,26 +1524,21 @@ private void DoAutoArchive(string fileName, LogEventInfo eventInfo)
...
+                return Path.ChangeExtension(fileName, ".{#}" + ext);
             }
             else
             {
                 //The archive file name is given. There are two possibilities
                 //(1) User supplied the Filename with pattern
                 //(2) User supplied the normal filename
-                fileNamePattern = this.ArchiveFileName.Render(eventInfo);
-                fileNamePattern = CleanupInvalidFileNameChars(fileNamePattern);
+                string archiveFileName = this.ArchiveFileName.Render(eventInfo);
```
Comment (304NotModified) [inline src/NLog/Targets/FileTarget.cs]: I like these improvements (narrowing scope) :+1:

File: tests/NLog.UnitTests/Targets/FileTargetTests.cs [hunk 3]
```diff
@@ -1562,7 +1562,7 @@ public void FileTarget_LogAndArchiveFilesWithSameName_ShouldArchive()
         }
 
         [Fact]
-        public void Single_Archive_File_Rolls_Correctly()
+        public void SingleArchiveFileRollsCorrectly()
```
Comment (304NotModified) [inline tests/NLog.UnitTests/Targets/FileTargetTests.cs]: don't mind the snake cases in the test names. If there are in the test cases, it's fine IMO

File: tests/NLog.UnitTests/Targets/FileTargetTests.cs [hunk 4]
```diff
@@ -1580,8 +1580,71 @@ public void Single_Archive_File_Rolls_Correctly()
 
                 SimpleConfigurator.ConfigureForTargetLogging(ft, LogLevel.Debug);
 
-                // we emit 5 * 250 *(3 x aaa + \n) bytes
-                // so that we should get a full file + 3 archives
+                // we emit 2 * 250 *(aaa + \n) bytes
```
Comment (304NotModified) [inline tests/NLog.UnitTests/Targets/FileTargetTests.cs]: what have changed here?

**Comment by 304NotModified** [comment]
hey @bhaeussermann, thanks for your contribution again! 

I just checked your code in a short time (not done yet), and I have added some code comments. We will review you code more in depth soon.

We are really pleased with your contributions! Nice findings and fixes! Also happy with your good explanation :) 
Would you like to join the [NLog team](https://github.com/orgs/NLog/people)? :)

File: tests/NLog.UnitTests/Targets/FileTargetTests.cs (hunk 4 above)
Comment (bhaeussermann) [inline tests/NLog.UnitTests/Targets/FileTargetTests.cs]: Firstly the comment was wrong (it was probably copied from somewhere else), so I fixed it.

File: tests/NLog.UnitTests/Targets/FileTargetTests.cs [hunk 5]
```diff
@@ -1580,8 +1580,71 @@ public void Single_Archive_File_Rolls_Correctly()
...
+                AssertFileContents(
+                    Path.Combine(tempPath, "archive", "file.txt2"),
+                    StringRepeat(250, "aaa\n"),
+                    Encoding.UTF8);
+
+                for (var i = 0; i < 250; ++i)
+                {
+                    logger.Debug("ccc");
+                }
+
```
Comment (bhaeussermann) [inline tests/NLog.UnitTests/Targets/FileTargetTests.cs]: Here I extended the test to check that the second archive is created.

File: tests/NLog.UnitTests/Targets/FileTargetTests.cs [hunk 6]
```diff
@@ -1580,8 +1580,71 @@ public void Single_Archive_File_Rolls_Correctly()
...
+                if (File.Exists(tempFile))
+                    File.Delete(tempFile);
+                if (Directory.Exists(tempPath))
+                    Directory.Delete(tempPath, true);
+            }
+        }
+
+        [Fact]
+        public void ArchiveFileRollsCorrectly()
+        {
```
Comment (bhaeussermann) [inline tests/NLog.UnitTests/Targets/FileTargetTests.cs]: This test exposes the bug reported in issue #1108

File: tests/NLog.UnitTests/Targets/FileTargetTests.cs [hunk 7]
```diff
@@ -1590,17 +1653,96 @@ public void Single_Archive_File_Rolls_Correctly()
...
+                if (File.Exists(tempFile))
+                    File.Delete(tempFile);
+                if (Directory.Exists(tempPath))
+                    Directory.Delete(tempPath, true);
+            }
+        }
+
+        [Fact]
+        public void ArchiveFileRollsCorrectly_ExistingArchives()
+        {
```
Comment (bhaeussermann) [inline tests/NLog.UnitTests/Targets/FileTargetTests.cs]: Also a new test.

**Comment by bhaeussermann** [comment]
> Would you like to join the NLog team? :)

Sure! :P

**Comment by 304NotModified** [comment]
Cool! Check your mail and https://gitter.im/NLog/devs

**Comment by 304NotModified** [comment]
Thanks! I has been merged!

**Pull request merged**